dungminhnguyen257/the-art-commerce
# (next-auth) implement authentication and authorization using NextAuth and GoogleAuth

## Summary of changes:
- Add authentication logic
- Add authorization logic
- Add Next Auth
- Add google auth adapter
- Modify db schema for auth feature

## Test steps:
- In Home, click log in and sign in with your Google account
- After signed in, navigate to My Cart
- Click Sign Out
- Navigate to My Cart and the app should ask you to sign in
- Sign in again
- My Cart page should load after successful sign-in
